pyquil/api/_qvm.py

```diff
@@ -590,11 +520,127 @@ def augment_program_with_memory_values(self, quil_program: Program) -> Program:
 
         return percolate_declares(p)
 
-    @_record_call
-    def reset(self) -> None:
-        """
-        Reset the state of the underlying QAM, and the QVM connection information.
-        """
-        super().reset()
-        forest_connection = ForestConnection()
-        self.connection = forest_connection
+
+TYPE_EXPECTATION = "expectation"
+TYPE_MULTISHOT = "multishot"
+TYPE_MULTISHOT_MEASURE = "multishot-measure"
+TYPE_WAVEFUNCTION = "wavefunction"
+
+
+def validate_noise_probabilities(noise_parameter: Optional[Tuple[float, float, float]]) -> None:
+    """
+    Is noise_parameter a valid specification of noise probabilities for depolarizing noise?
+
+    :param tuple noise_parameter: Tuple of 3 noise parameter values to be validated.
+    """
+    if not noise_parameter:
+        return
+    if not isinstance(noise_parameter, tuple):
+        raise TypeError("noise_parameter must be a tuple")
+    if any([not isinstance(value, float) for value in noise_parameter]):
+        raise TypeError("noise_parameter values should all be floats")
+    if len(noise_parameter) != 3:
+        raise ValueError("noise_parameter tuple must be of length 3")
+    if sum(noise_parameter) > 1 or sum(noise_parameter) < 0:
+        raise ValueError("sum of entries in noise_parameter must be between 0 and 1 (inclusive)")
+    if any([value < 0 for value in noise_parameter]):
+        raise ValueError("noise_parameter values should all be non-negative")
+
+
+def validate_qubit_list(qubit_list: Sequence[int]) -> Sequence[int]:
+    """
+    Check the validity of qubits for the payload.
```

Comment:
Maybe note what the criteria are

Reply:
I should have mentioned: this and the two functions around it were simply moved over from `_base
_connection.py`, so I don't know a ton about their implementation